modules/gapi/test/internal/gapi_int_executor_tests.cpp

```diff
@@ -6,10 +6,158 @@
 
 
 #include "../test_precomp.hpp"
+#include "../gapi_mock_kernels.hpp"
 
 namespace opencv_test
 {
 
+namespace
+{
+
+class GMockExecutable final: public cv::gimpl::GIslandExecutable
```

Comment:
holly molly

Reply:
..thats brute but good.

My concern is that this class may expose way too much details of our internals here in the test level, so that if internal API changes, we'll have to update (or NOT forget to update) this thing as well

Reply:
Honestly I don't know how to test it properly. From my point of view this approach allows us to be sure that everything works fine.

>>> we'll have to update (or NOT forget to update
Yep, agreed, but I we should be ready for that since we have such `internal` tests suit.

What's the final decision here ?

Reply:
keep as is